### PR TITLE
Updated Update-AzureModule script

### PR DESCRIPTION
**Pull request by D1v38om83r**
Updated Update-AzureModule script to use ModuleVersionsOverrides parameter

### CODE REVIEW

File: Utility/ARM/Update-AzureModule.ps1 [hunk 1]
```diff
@@ -71,7 +111,23 @@ try
...
+        $Body = @"
+            {
+               "properties":{
+               "runbook":{
+                   "name":"Update-AutomationAzureModulesForAccount"
+               },
+               "parameters":{
+                    "ResourceGroupName":"$AutomationResourceGroup",
+                    "AutomationAccountName":"$AutomationAccount",
+                    "ModuleVersionOverrides": $versionOverrides
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Don't we need double quotes around this? From the runbook invocation perspective, we are supposed to pass a string here, not a JSON object

File: Utility/ARM/Update-AzureModule.ps1 [hunk 2]
```diff
@@ -28,8 +40,36 @@ Param
...
+        $versionOverrides = ConvertTo-Json $ModuleVersionOverrides
+    } elseif ($ModuleVersionOverrides.GetType() -eq ("").GetType()){
+        ## ModuleVersionOverrides is of type String
+        ## Verify that the ModuleVersionOverrides can be deserialized
+        try{
+            $temp = ConvertFrom-Json $ModuleVersionOverrides
+        }
+        catch [System.ArgumentException]{
+            $ex = $_ 
+		    ## rethrow intended
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Indentation seems to be off.

File: Utility/ARM/Update-AzureModule.ps1 (hunk 1 above)
Comment (D1v38om83r) [reply Utility/ARM/Update-AzureModule.ps1]: yes, that is correct, thank you for pointing that out. Fixed in next commit

File: Utility/ARM/Update-AzureModule.ps1 [hunk 3]
```diff
@@ -28,8 +40,36 @@ Param
...
+        $versionOverrides = ConvertTo-Json $ModuleVersionOverrides
+    } elseif ($ModuleVersionOverrides.GetType() -eq ("").GetType()){
+        ## ModuleVersionOverrides is of type String
+        ## Verify that the ModuleVersionOverrides can be deserialized
+        try{
+            $temp = ConvertFrom-Json $ModuleVersionOverrides -ErrorAction Stop
+        }
+        catch [System.ArgumentException]{
+            $ex = $_ 
+		    ## rethrow intended
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Indentation is off

File: Utility/ARM/Update-AzureModule.ps1 [hunk 4]
```diff
@@ -28,8 +40,36 @@ Param
...
     [Parameter(Mandatory=$True)]
-    [String] $AutomationAccount
+    [String] $AutomationAccount,
+
+    [Parameter(Mandatory=$False)]
+    [object] $ModuleVersionOverrides
     )
+
+$versionOverrides = ""
+## Try to parse module version overrides
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Any specific reason to use double # for comments? This makes me think these comments have a special meaning. If they don't, let's use just a single #.

File: Utility/ARM/Update-AzureModule.ps1 [hunk 5]
```diff
@@ -28,8 +40,36 @@ Param
...
+if ($ModuleVersionOverrides){
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Minor: please add a space before {, here and below.

File: Utility/ARM/Update-AzureModule.ps1 [hunk 6]
```diff
@@ -28,8 +40,36 @@ Param
...
+    if ($ModuleVersionOverrides.GetType() -eq (@{}).GetType()){
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Consider "-eq [hashtable]". The result is the same, but looks a bit more readable.

File: Utility/ARM/Update-AzureModule.ps1 [hunk 7]
```diff
@@ -28,8 +40,36 @@ Param
...
+        ## ModuleVersionOverrides is of type HashTable
+        $versionOverrides = ConvertTo-Json $ModuleVersionOverrides
+    } elseif ($ModuleVersionOverrides.GetType() -eq ("").GetType()){
```
Comment (AnatoliB) [inline Utility/ARM/Update-AzureModule.ps1]: Consider "-eq [string]"

**Pull request merged**